pkg/ee/cluster-backup/resources/seed-cluster/deployment.go

```diff
@@ -0,0 +1,238 @@
+			}
+
+			dep.Spec.Template.ObjectMeta = metav1.ObjectMeta{
+				Labels: podLabels,
+				Annotations: map[string]string{
+					"prometheus.io/path":   "/metrics",
+					"prometheus.io/port":   "8085",
+					"prometheus.io/scrape": "true",
+				},
+			}
+
+			dep.Spec.Template.Spec.Volumes = volumes
+			dep.Spec.Template.Spec.InitContainers = []corev1.Container{
+				{
+					Name:  "velero-velero-plugin-for-aws",
+					Image: "velero/velero-plugin-for-aws:v1.0.0",
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "plugins",
+							MountPath: "/target",
+						},
+					},
+					ImagePullPolicy: corev1.PullPolicy("IfNotPresent"),
+				},
+			}
+
+			dep.Spec.Template.Spec.Containers = []corev1.Container{
+				{
+					Name:    "velero",
+					Image:   veleroImage,
+					Command: []string{"/velero"},
+					Args: []string{
+						"server",
+						"--features=",
+						"--uploader-type=kopia",
+						fmt.Sprintf("--namespace=%s", resources.ClusterBackupNamespaceName),
+						"--kubeconfig=/etc/kubernetes/kubeconfig/kubeconfig",
+					},
+					Ports: []corev1.ContainerPort{
+						{
+							Name:          "metrics",
+							ContainerPort: 8085,
+							Protocol:      corev1.ProtocolSCTP,
+						},
+					},
+					Env: []corev1.EnvVar{
+						{
+							Name:  "VELERO_SCRATCH_DIR",
+							Value: "/scratch",
+						},
+						{
+							Name:  "LD_LIBRARY_PATH",
+							Value: "/plugins",
+						},
+						{
+							Name:  "GOOGLE_APPLICATION_CREDENTIALS",
+							Value: "/credentials/cloud",
+						},
+						{
+							Name:  "AWS_SHARED_CREDENTIALS_FILE",
+							Value: "/credentials/cloud",
+						},
+						{
+							Name:  "AZURE_CREDENTIALS_FILE",
+							Value: "/credentials/cloud",
+						},
+						{
+							Name:  "ALIBABA_CLOUD_CREDENTIALS_FILE",
+							Value: "/credentials/cloud",
+						},
+						// looks like velero has a bug where it doesn't use the provided kubeconfig for some operations
+						// and falls back to in-cluster credentials. This is a workaround.
+						{
+							Name:  "KUBECONFIG",
+							Value: "/etc/kubernetes/kubeconfig/kubeconfig",
+						},
+					},
+					VolumeMounts:    volumeMounts,
+					ImagePullPolicy: corev1.PullPolicy("IfNotPresent"),
+				},
+			}
+			dep.Spec.Template.Spec.ServiceAccountName = "default"
+			err = resources.SetResourceRequirements(dep.Spec.Template.Spec.Containers, nil, nil, dep.Annotations)
+			if err != nil {
+				return nil, fmt.Errorf("failed to set resource requirements: %w", err)
+			}
+
+			return dep, nil
+		}
+	}
+}
+
+func getVolumeMounts() []corev1.VolumeMount {
+	return []corev1.VolumeMount{
+		{
+			Name:      resources.ClusterbackupKubeconfigSecretName,
+			MountPath: "/etc/kubernetes/kubeconfig",
+			ReadOnly:  true,
+		},
+		{
+			Name:      resources.CASecretName,
+			MountPath: "/etc/kubernetes/pki/ca",
+			ReadOnly:  true,
+		},
+		{
+			Name:      resources.CABundleConfigMapName,
+			MountPath: "/etc/kubernetes/pki/ca-bundle",
+			ReadOnly:  true,
+		},
```

Comment:
(How) is this CABundle being used? Does Velero automagically find it at this location?